engine/src/main/java/org/hibernate/search/query/dsl/impl/ConnectedMultiFieldsRangeQueryBuilder.java

```diff
@@ -116,12 +116,16 @@ private static Query createKeywordRangeQuery(String fieldName, RangeQueryContext
 		final Object fromObject = rangeContext.getFrom();
 		final Object toObject = rangeContext.getTo();
 		final DocumentBuilderIndexedEntity documentBuilder = Helper.getDocumentBuilder( queryContext );
-		final String fromString = fieldContext.objectToString( documentBuilder, fromObject, conversionContext );
+		final String fromString = fromObject == null ?
+				null :
+				fieldContext.objectToString( documentBuilder, fromObject, conversionContext );
 		final String lowerTerm = fromString == null ?
 				null :
 				Helper.getAnalyzedTerm( fieldName, fromString, "from", queryAnalyzer, fieldContext );
 
-		final String toString = fieldContext.objectToString( documentBuilder, toObject, conversionContext );
+		final String toString = toObject == null ?
```

Comment:
My feeling is ```below(null)``` and ```above(null)``` should be forbidden. Because technically only one of the bounds of ```TermRangeQuery``` can be null, not both. We are making an error by using the null token to fill in. It should not play any role in range queries because it is only a marker so it only makes sense for == or "is null' type of comparisons. 

There is also a deeper reason to disallow it. Because such a query would either match all or nothing, so it would be pointless. I support my theory by looking at how others handle this, specifically JPA and SQL. Here is a piece of jpa spec regarding the results of 'order by':

>SQL rules for the ordering of null values apply: that is, all null values must appear before all non-null values in the ordering or all null values must appear after all non-null values in the ordering, but it is not specified which.